### ISSUE TITLE
Converted Lung Dragons expecting xnb icon

### ISSUE THREAD

**Issue by htgiddings** [AnotherPillow/BFAV2CP#5]
Using this utility I converted [BFAV Lung Dragons](https://www.nexusmods.com/stardewvalley/mods/5761) to a Content Patcher version and dropped it in my mods. When trying to purchase animals the livestock menu never opens, I just go back to Marnie's initial dialog no matter how I select the option to purchase animals. Her supply shop and the "leave" option work just fine. Below is the stack trace I'm getting every time I select that option:
```[game] An error occurred in the base update loop: Microsoft.Xna.Framework.Content.ContentLoadException: The content file was not found.
 ---> FileNotFoundException: Could not find file '/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/Animals/stormdrakon.BFAVLungDragons-River Lung Icon.xnb'.
File name: '/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/Animals/stormdrakon.BFAVLungDragons-River Lung Icon.xnb'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Microsoft.Xna.Framework.TitleContainer.PlatformOpenStream(String safeName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Platform\TitleContainer.Desktop.cs:line 29
   at Microsoft.Xna.Framework.TitleContainer.OpenStream(String name) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\TitleContainer.cs:line 37
   at Microsoft.Xna.Framework.Content.ContentManager.OpenStream(String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 272
   --- End of inner exception stack trace ---
   at Microsoft.Xna.Framework.Content.ContentManager.OpenStream(String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 289
   at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](String assetName, Action`1 recordDisposableObject) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 319
   at Microsoft.Xna.Framework.Content.ContentManager.Load[T](String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 255
   at StardewModdingAPI.Framework.ContentManagers.BaseContentManager.RawLoad[T](IAssetName assetName, Boolean useCache) in SMAPI\Framework\ContentManagers\BaseContentManager.cs:line 340
   at StardewModdingAPI.Framework.ContentManagers.GameContentManager.<>c__DisplayClass6_0`1.<LoadExact>b__0() in SMAPI\Framework\ContentManagers\GameContentManager.cs:line 127
   at StardewModdingAPI.Framework.Utilities.ContextHash`1.Track[TResult](T key, Func`1 action) in SMAPI\Framework\Utilities\ContextHash.cs:line 53
   at StardewModdingAPI.Framework.ContentManagers.GameContentManager.LoadExact[T](IAssetName assetName, Boolean useCache)
   at StardewModdingAPI.Framework.ContentManagers.BaseContentManager.LoadLocalized[T](IAssetName assetName, LanguageCode language, Boolean useCache) in SMAPI\Framework\ContentManagers\BaseContentManager.cs:line 154
   at StardewValley.Menus.PurchaseAnimalsMenu..ctor(List`1 stock, GameLocation targetLocation) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Menus\PurchaseAnimalsMenu.cs:line 74
   at StardewValley.GameLocation.<>c__DisplayClass442_0.<ShowAnimalShopMenu>b__0(String value) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12153
   at StardewValley.GameLocation.ShowPagedResponses(String prompt, List`1 responses, Action`1 on_response, Boolean auto_select_single_choice, Boolean addCancel, Int32 itemsPerPage) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12036
   at StardewValley.GameLocation.ShowAnimalShopMenu(Action`1 onMenuOpened) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12145
   at StardewValley.GameLocation.answerDialogueAction_PatchedBy<spacechase0.SpaceCore>(GameLocation this, String questionAndAnswer, String[] questionParams)
   at StardewValley.Menus.DialogueBox.receiveLeftClick(Int32 x, Int32 y, Boolean playSound)
   at StardewValley.Game1.updateActiveMenu(GameTime gameTime)
   at StardewValley.Game1._update(GameTime gameTime)
   at StardewValley.Game1.Update(GameTime gameTime)
   at StardewModdingAPI.Framework.SCore.OnPlayerInstanceUpdating(SGame instance, GameTime gameTime, Action runUpdate)
[game] WARNING: Mismatched UI Mode Push/Pop counts. Correcting.
```

I'm noticing two things off the bat:
1. There are no xnb files in the converted assets, only pngs.
2. The path to the Lung Dragons from the mod root folder (/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/) is "Animals/New Livestock/[CP] Lung Dragons 2.0", not "Animals"

Environment Information:
OS: MacOS 14.3.1
SDV: 1.6.5
SMAPI: 4.0.8
Python: 3.12

**Comment by AnotherPillow** [comment]
Output log? Mod link? Screenshot of output mod folder?On 22 Apr 2024 14:16, Henry Giddings ***@***.***> wrote:
Using this utility I converted BFAV Lung Dragons to a Content Patcher version and dropped it in my mods. When trying to purchase animals the livestock menu never opens, I just go back to Marnie's initial dialog no matter how I select the option to purchase animals. Her supply shop and the "leave" option work just fine. Below is the stack trace I'm getting every time I select that option:
 ---> FileNotFoundException: Could not find file '/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/Animals/stormdrakon.BFAVLungDragons-River Lung Icon.xnb'.
File name: '/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/Animals/stormdrakon.BFAVLungDragons-River Lung Icon.xnb'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Microsoft.Xna.Framework.TitleContainer.PlatformOpenStream(String safeName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Platform\TitleContainer.Desktop.cs:line 29
   at Microsoft.Xna.Framework.TitleContainer.OpenStream(String name) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\TitleContainer.cs:line 37
   at Microsoft.Xna.Framework.Content.ContentManager.OpenStream(String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 272
   --- End of inner exception stack trace ---
   at Microsoft.Xna.Framework.Content.ContentManager.OpenStream(String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 289
   at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](String assetName, Action`1 recordDisposableObject) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 319
   at Microsoft.Xna.Framework.Content.ContentManager.Load[T](String assetName) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\MonoGame.Desktop\MonoGame.Framework\Content\ContentManager.cs:line 255
   at StardewModdingAPI.Framework.ContentManagers.BaseContentManager.RawLoad[T](IAssetName assetName, Boolean useCache) in SMAPI\Framework\ContentManagers\BaseContentManager.cs:line 340
   at StardewModdingAPI.Framework.ContentManagers.GameContentManager.<>c__DisplayClass6_0`1.<LoadExact>b__0() in SMAPI\Framework\ContentManagers\GameContentManager.cs:line 127
   at StardewModdingAPI.Framework.Utilities.ContextHash`1.Track[TResult](T key, Func`1 action) in SMAPI\Framework\Utilities\ContextHash.cs:line 53
   at StardewModdingAPI.Framework.ContentManagers.GameContentManager.LoadExact[T](IAssetName assetName, Boolean useCache)
   at StardewModdingAPI.Framework.ContentManagers.BaseContentManager.LoadLocalized[T](IAssetName assetName, LanguageCode language, Boolean useCache) in SMAPI\Framework\ContentManagers\BaseContentManager.cs:line 154
   at StardewValley.Menus.PurchaseAnimalsMenu..ctor(List`1 stock, GameLocation targetLocation) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Menus\PurchaseAnimalsMenu.cs:line 74
   at StardewValley.GameLocation.<>c__DisplayClass442_0.<ShowAnimalShopMenu>b__0(String value) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12153
   at StardewValley.GameLocation.ShowPagedResponses(String prompt, List`1 responses, Action`1 on_response, Boolean auto_select_single_choice, Boolean addCancel, Int32 itemsPerPage) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12036
   at StardewValley.GameLocation.ShowAnimalShopMenu(Action`1 onMenuOpened) in D:\GitlabRunner\builds\Gq5qA5P4\1\ConcernedApe\stardewvalley\Farmer\Farmer\Locations\GameLocation.cs:line 12145
   at StardewValley.GameLocation.answerDialogueAction_PatchedBy<spacechase0.SpaceCore>(GameLocation this, String questionAndAnswer, String[] questionParams)
   at StardewValley.Menus.DialogueBox.receiveLeftClick(Int32 x, Int32 y, Boolean playSound)
   at StardewValley.Game1.updateActiveMenu(GameTime gameTime)
   at StardewValley.Game1._update(GameTime gameTime)
   at StardewValley.Game1.Update(GameTime gameTime)
   at StardewModdingAPI.Framework.SCore.OnPlayerInstanceUpdating(SGame instance, GameTime gameTime, Action runUpdate)
[game] WARNING: Mismatched UI Mode Push/Pop counts. Correcting.

I'm noticing two things off the bat:
There are no xnb files in the converted assets, only pngs.The path to the Lung Dragons from the mod root folder (/Users/gaming/Library/Application Support/Steam/steamapps/common/Stardew Valley/Contents/Resources/Content/) is "Animals/New Livestock/[CP] Lung Dragons 2.0", not "Animals"
Environment Information:
OS: MacOS 14.3.1
SDV: 1.6.5
SMAPI: 4.0.8
Python: 3.12

**Comment by htgiddings** [comment]
1. I linked to the mod in the first sentence of my comment
2. Didn't save the output. Didn't see any errors though.
3. <img width="245" alt="Screenshot 2024-04-22 at 7 16 38 PM" src="https://github.com/AnotherPillow/BFAV2CP/assets/17188267/2efce0c3-5740-493b-b366-d1bc11b2a52b">

**Comment by AnotherPillow** [comment]
Then rerun it?On 23 Apr 2024 00:20, Henry Giddings ***@***.***> wrote:
I linked to the mod in the first sentence of my commentDidn't save the output. Didn't see any errors though.
Screenshot.2024-04-22.at.7.16.38.PM.png (view on web)

**Comment by htgiddings** [comment]
No, I’m just gonna grab the official update.On Apr 23, 2024, at 2:26 AM, AnotherPillow ***@***.***> wrote:﻿
Then rerun it?On 23 Apr 2024 00:20, Henry Giddings ***@***.***> wrote:

I linked to the mod in the first sentence of my commentDidn't save the output. Didn't see any errors though.

Screenshot.2024-04-22.at.7.16.38.PM.png (view on web)

**Comment by htgiddings** [comment]
Yeah this wasn’t complete, whatever the issue was you haven’t provided any evidence that you even looked into it.On May 25, 2024, at 9:02 AM, AnotherPillow ***@***.***> wrote:﻿
Closed #5 as completed.

**Comment by AnotherPillow** [comment]
> Yeah this wasn’t complete, whatever the issue was you haven’t provided any evidence that you even looked into it.On May 25, 2024, at 9:02 AM, AnotherPillow ***@***.***> wrote:﻿ Closed #5 as completed. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

You said that you weren't even going to use it. I ran that mod myself, it worked fine. You cannot expect me to know what a bug is without **EVEN BOTHERING TO SHOW THE ERROR**.
